sregistry/main/registry/auth.py

```diff
@@ -16,25 +16,45 @@
     generate_timestamp
 )
 
+from sregistry.logger import bot
+
 def authorize(self, names, payload=None, request_type="push"):
     '''Authorize a client based on encrypting the payload with the client
        token, which should be matched on the receiving server'''
 
-    if self.secrets is not None:
+    self.username = self._get_and_update_setting('SREGISTRY_REGISTRY_USERNAME')
```

Comment:
I'm seeing a lot of duplicate code here... can we create a shared function?

Reply:
But if you like as is, I don't feel super strong about this. Someone else can complain and change it later. The function is fairly small still and it's only repeated once, so I'm okay if you want to leave as is.

Reply:
Have refactored it a bit... think it's neater now.